src/zope/sendmail/delivery.py

```diff
@@ -138,7 +138,7 @@ def send(self, fromaddr, toaddrs, message):
             messageid = messageid[1:-1]
         else:
             messageid = self.newMessageId()
-            message = b'Message-Id: <%s>\n%s' % (messageid.encode(), message)
+            message = b'Message-Id: <%s>\r\n%s' % (messageid.encode(), message)
```

Comment:
Conceptionally, this change is suboptimal: a few lines above, the line separator is determined assuming consistency; in this line the consistency may be broken.

I suggest the following change: ensure that `line_sep`  gets defined even for `message is None`; use `line_sep` in the construction of the `Message-Id` header.
This would preserve consistency and delegate the use of the "correct" line separator to the caller.

Alternatively, we could convert wrong line endings to `\r\n' here. But I do not think we should do that: the task of `deliver` is to deliver a message not to change it. Apparently, the message id is required during later processing, thus it might be acceptable to add it -- provided that the addition preserves consistency.

Reply:
Take a look at my last changes, I believe that's what you were looking for, correct?